Sources/Purchasing/EntitlementInfo.swift

```diff
@@ -282,23 +282,41 @@ public extension EntitlementInfo {
 
 }
 
+// MARK: - Internal
+
+extension EntitlementInfo {
+
+    /// This grace period allows apps to continue functioning if the backend is down, but for a limited time.
+    /// We don't want to continue granting entitlements with an outdated `requestDate` forever,
+    /// since that would allow a user to get a free trial, then go offline and keep the entitlement with no time limit.
+    static let requestDateGracePeriod: DispatchTimeInterval = .days(3)
+
+}
+
 // MARK: - Private
 
 private extension EntitlementInfo {
 
-    class func isDateActive(expirationDate: Date?, forRequestDate requestDate: Date?) -> Bool {
+    static func isDateActive(expirationDate: Date?, for requestDate: Date?) -> Bool {
```

Comment:
Also, while working on the Android side, I noticed we could assume we always have a `requestDate`, so this wouldn't be nullable. I think it would be the same in iOS, but not sure if I'm missing something.

Reply:
I think you're right!

Reply:
#2325